sycl/include/sycl/ext/intel/experimental/esimd/memory.hpp

```diff
@@ -582,11 +593,20 @@ lsc_block_load(AccessorTy acc, uint32_t offset,
       detail::to_lsc_vector_size<NElts / SmallIntFactor>();
 
   if constexpr (SmallIntFactor == 1) {
-    return __esimd_lsc_load_bti<T, L1H, L3H, _AddressScale, _ImmOffset, _DS,
-                                _VS, _Transposed, N>(pred.data(),
-                                                     offsets.data(), si);
+    if constexpr (_DS == lsc_data_size::u32) {
+      __ESIMD_NS::simd<uint32_t, NElts> result =
+          __esimd_lsc_load_bti<uint32_t, L1H, L3H, _AddressScale, _ImmOffset,
+                               lsc_data_size::u32, _VS, _Transposed, N>(
+              pred.data(), offsets.data(), si);
+      return result.template bit_cast_view<T>();
```

Comment:
@kbobrovs - accordingly to your experience, is this bit_cast_view() always optimized away if the type is already integral?
If not, then it probably makes sense to write something like:
```
if constexpr (std::is_integral_v<T>) {
    return __esimd_lsc_load_bit<....>(....); 
} else {
    result = __esimd_lsc_load_bit<uint..., >(....);
    return result.template bit_cast_view<T>();
}
```
What is your recommendation here?

Reply:
It is optimized, but I'd prefer `sycl::bit_cast` instead in such cases.